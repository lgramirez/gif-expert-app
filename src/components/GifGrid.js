import React, { useEffect, useState } from "react";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url =
      "https://api.giphy.com/v1/gifs/search?q=rick+and+morthy&limit=10&api_key=G3x1V4VNTyP4snSskhcKfu11XFgJbUSw";
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images.downsized_medium.url,
      };
    });

    console.log(gifs);
    setImages(gifs);
  };

  return (
    <>
      <h3>{category}</h3>

      {images.map((image) => {
        return <GifGridItem key={image.id} {...image} />;
      })}
    </>
  );
};
