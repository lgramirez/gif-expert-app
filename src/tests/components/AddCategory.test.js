import React from "react";
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("Pruebas para <AddCategory />", () => {
  // crea una funcion simulada nueva y sin uso
  const setCategories = jest.fn();

  // hacemos una doble inicializacion para tener las ayudas para wrapper
  // por ejm wrapper.find()
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    // limpia todas las simulaciones, especialmente para la funcion que creamos
    // de setCategories
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("Debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola Mundo";

    input.simulate("change", { target: { value } });

    // para hacer este expect tuvimos que a√±adir un parrafo <p></p>
    // al componente, actualizar el snapshot y comprobar que el
    // parrafo se actualice con el valor del valor que cambiamos en el input
    expect(wrapper.find("p").text().trim()).toBe(value);
  });

  test("No debe de postear la informacion con submit", () => {
    // creamos la funcion preventDefault de una manera corta
    // sino tambien podria ser preventDefault: () => {}
    // cualquier cosa revisar como crear un metodo de un objeto js
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    // con el metodo haveBeenCalled preguntamos si la funcion simulada
    // creada por jest fue ejecutada
    expect(setCategories).not.toHaveBeenCalled();
  });

  // tarea
  test("Debe de llamar el setCategories y limpiar la caja de texto", () => {
    const value = "Hola mundo cruel";

    // 1. simular el inputChange
    const input = wrapper.find("input");
    input.simulate("change", { target: { value } });

    // 2. simular el submit
    const form = wrapper.find("form");
    form.simulate("submit", { preventDefault: () => {} });

    // 3. setCategories se debe de haber llamado
    expect(setCategories).toHaveBeenCalled();

    // 4. el valor del input debe de estar ''
    expect(input.prop("value")).toBe("");
  });
});
