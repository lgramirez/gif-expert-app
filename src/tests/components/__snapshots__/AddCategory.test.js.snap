// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <AddCategory /> debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
